### ISSUE TITLE
JSON v0.3

### ISSUE THREAD

**Issue by ValerioLyndon** [ValerioLyndon/Theme-Customiser#41]
JSON v0.3 is overdue as I am already finding usability annoyances with 0.2.

A non-comprehensive list of changes to be made are as follows:

- Split keys relating to recommended theme configuration into two duplicate objects. One being a "preview_config" object and the other being a "user_config" or "recommended_config" object (name to be decided). The preview_config would inherit settings from the user_config, but not the other way around. This would provide a powerful way to display your theme without having it affect the recommended install settings, or vice versa. As it stands now, if a theme has an optimal display of no cover image, or certain list columns, there is no way to make it look its best in the preview without changing the recommended installation for the user at the same time. An example of the new JSON:
```
[
  "theme": "Example",
  "config": {
    "columns": {
      "animelist": {
        "Image": null,
        "Score": false
      }
    }
    "cover": null,
    "style": [1,2]
  },
  "preview_config": {
    "columns": {
      "animelist": {
        "Image": true
      }
    }
  }
]
```
- Convert replacements from an array to a dictionary with various keys: "find", "fill", and a third. Perhaps "off_fill"? "toggle_fill"? Or perhaps use 3 replacements for all entries, with the "off" state being when it matches the default. Maybe it should be "fill_on" and "fill_off"? "fill_false" and "fill_true"? Something like that.

These changes are open to discussion and not extensive. It should be noted that all these changes should be easily convertable from the old format once I've worked more on #22 which should contain an auto-update for JSON.

**Comment by shaggyze** [comment]
I get the gist of what you're proposing, but it does sound confusing, not sure if many will need/use it, but I think in the long run it's a good addition for advanced customization between user, preview, and recommended.

**Comment by ValerioLyndon** [comment]
I agree it could be a tad confusing, but I think once I've written a proper guide for it, it shouldn't be too bad. Basically the two "config" keys have the exact same features and the only difference is that one informs what the user should do to install the theme, and the other informs the preview what settings to display for the best first impression.

The reason I want to make this distinction is because I have several themes that *can* work with various list columns, so of course I list those columns as optional. But some combinations or selections look better than others and are more intended, so in the preview I wish to display a specific set for users to see for the first time.

Take Clarity, for instance. It can work with or without images and tags, but I believe it looks best with them enabled and the image & tag-related mods need them enabled for a proper preview. But with the current system, I can't force these columns to be enabled without simultaneously changing the recommended install instructions, which I don't want to do. The only way around this is to have two different "columns" keys. 

Of course, no one is required to make use of this feature. The basic JSON will still be pretty simple. But it gives more ability to the JSON to make everything look it's best and inform the end user to the fullest.

**Comment by ValerioLyndon** [comment]
The one thing I'm unsure of is the need for the new "config" key. I could probably just keep the current structure as-is and simply use the "preview" key as an optional overwrite. I think it may be a bit cleaner to keep things "flatter", without the extra key. But it also makes sense to have the extra "config" key to make it very clear which items can or can't be used inside of the "preview_config" key.

Code examples:
```
"columns": {…}
"preview": {
	"columns": {…}
}
```
As opposed to:
```
"config": {
	"columns": {…}
},
"preview_config": {
	"columns": {…}
}
```

**Comment by ValerioLyndon** [comment]
Upon further thought, I've delayed the replacement changes and have kept the current position of the "columns" key. This means that no new version is required as-of-yet since there will not be any breaking changes. The other changes mentioned will be implemented in PR #42. Closing for now, will re-open if/when the need arises.
